src/librustc_mir/interpret/intrinsics.rs

```diff
@@ -46,8 +46,11 @@ crate fn eval_nullary_intrinsic<'tcx>(
     def_id: DefId,
     substs: SubstsRef<'tcx>,
 ) -> InterpResult<'tcx, &'tcx ty::Const<'tcx>> {
-    let tp_ty = substs.type_at(0);
     let name = &*tcx.item_name(def_id).as_str();
+    if name == "is_const_eval" {
+        return Ok(ty::Const::from_bool(tcx, true));
+    }
+    let tp_ty = substs.type_at(0);
```

Comment:
Move this into the arms that need it and add the new intrinsic as its own arm below

Reply:
All arms need it :/

Reply:
Hmm. I guess there are no other nullary intrinsics without generics.

Reply:
This is still very ugly code though. :/ There should at least be comments explaining why the structure is so odd.